### PR TITLE
Add swiftui support

### PR DESCRIPTION
**Pull request by petar-jadek**
## Summary

The main Collar Logs screen has undergone a complete redesign. It has been upgraded to SwiftUI, preserving its core mechanics while introducing new features. Users can now copy log information, delete specific logs, or all of them. Additionally, writing search text will highlight matching log text for easier lookup. The LogItemPopupQueue has been removed as it was rarely used. The example app has been adjusted to work with both SwiftUI and UIKit, while old assets have been removed and replaced with SFSymbols.

**Related issue**: None
Productive task: https://app.productive.io/1-infinum/tasks/10555170

## Changes

- New SwiftUI Collar Logs View
- Removed UIKit components for showing Collar logs
- Refactored UIKit showLogs method to display new Collar Logs View
- Added modifier support showCollarLogs(isPresented: Binding<Bool>)

### Type

- [x] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [x] **Refactor**: This pull request refactors existing code.
- [x] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

This should not be breaking change since the way logs are being collected and displayed is not changed in any way. 

### Description

Demo of the new changes

https://github.com/user-attachments/assets/3d08e97b-1538-4beb-bac2-1374ecbfe0da

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [x] I have updated the documentation to reflect my changes (if applicable).

## Additional notes

None

### CODE REVIEW

File: Example/Collar/CollarView.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+//  Created by Petar Jadek on 08.10.2025.
+//  Copyright (c) 2020 Petar Jadek. All rights reserved.
+//
+
+import SwiftUI
+import Collar
+
+struct CollarView: View {
+
+    private let analyticsCollectionManager = AnalyticsCollectionManager.shared
```
Comment (Antonijo2307) [inline Example/Collar/CollarView.swift]: Remove this line if it you are not using it

File: Example/Collar/ViewController.swift [hunk 2]
```diff
@@ -7,27 +7,28 @@
...
     override func viewDidLoad() {
         super.viewDidLoad()
-        
-        LogItemPopupQueue.shared.enabled = true
-        LogItemPopupQueue.shared.showOnView = { UIApplication.shared.keyWindow }
+
+        view.backgroundColor = .systemBackground
+        setupButtons()
 
         DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
```
Comment (Antonijo2307) [inline Example/Collar/ViewController.swift]: Why do we need this delays? Could this be done by calling the log on some other event?

File: Sources/Collar/Classes/List/SwiftUI/LogListView.swift [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    var filteredItems: [LogItem] {
+        items.filter { item in
+            let matchesSearch: Bool
+            if searchText.isEmpty {
+                matchesSearch = true
+            } else {
+                matchesSearch =
+                item.name.localizedCaseInsensitiveContains(searchText) ||
+                (item.subtitleDisplay ?? "").localizedCaseInsensitiveContains(searchText)
+            }
```
Comment (kjaklinovic) [inline Sources/Collar/Classes/List/SwiftUI/LogListView.swift]: New swift syntax should allow something like this
```suggestion
            let matchesSearch: Bool = if searchText.isEmpty {
                true
            } else {
                item.name.localizedCaseInsensitiveContains(searchText) ||
                (item.subtitleDisplay ?? "").localizedCaseInsensitiveContains(searchText)
            }
```

File: Sources/Collar/Classes/List/SwiftUI/LogListView.swift [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+            let matchesFilter: Bool
+            if selectedFilterType == .all {
+                matchesFilter = true
+            } else {
+                matchesFilter = selectedFilterType.logType.contains(item.type)
+            }
```
Comment (kjaklinovic) [inline Sources/Collar/Classes/List/SwiftUI/LogListView.swift]: Same here.

**Review by kjaklinovic — APPROVED** [review]
This looks good to me on first glance besides few minor code improvements.

**Review by Antonijo2307 — APPROVED** [review]
